Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_PlayerClimb.cs

```diff
@@ -0,0 +1,188 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+    using System.Collections;
+
+    public struct PlayerClimbEventArgs
+    {
+        public uint controllerIndex;
+        public GameObject target;
+    }
+
+    public delegate void PlayerClimbEventHandler(object sender, PlayerClimbEventArgs e);
+
+    public class VRTK_PlayerClimb : MonoBehaviour
+    {
+        public event PlayerClimbEventHandler PlayerClimbStarted;
+        public event PlayerClimbEventHandler PlayerClimbEnded;
+
+        public bool usePlayerScale = true;
+        public bool useGravity = true;
+
+        private Transform controllerTransform;
+        private Vector3 startControllerPosition;
+        private Vector3 startPosition;
+
+        private Vector3 lastGoodHeadsetPosition;
+        private bool headsetColliding = false;
+        private bool isClimbing = false;
+
+        VRTK_PlayerPhysics playerPhysics;
+        private bool lastGravitySetting;
+
+        public virtual void OnPlayerClimbStarted(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbStarted != null)
+                PlayerClimbStarted(this, e);
+        }
+
+        public virtual void OnPlayerClimbEnded(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbEnded != null)
+                PlayerClimbEnded(this, e);
+        }
+
+        protected PlayerClimbEventArgs SetPlayerClimbEvent(uint controllerIndex, GameObject target)
+        {
+            PlayerClimbEventArgs e;
+            e.controllerIndex = controllerIndex;
+            e.target = target;
+            return e;
+        }
+
+        private void Start()
+        {
+            var controllerManager = GameObject.FindObjectOfType<SteamVR_ControllerManager>();
```

Comment:
I dont think you need to specify GameObject, you can just do:

`var controllerManager = FindObjectOfType<SteamVR_ControllerManager>();`

Visual Studio should highlight it and moan that an unnecessary keyword is there.,